dialectic-macro/src/lib.rs

```diff
@@ -338,14 +358,35 @@ impl Parse for OfferInvocation {
 
 impl ToTokens for OfferOutput {
     fn to_tokens(&self, tokens: &mut TokenStream) {
+        use proc_macro_crate::FoundCrate;
+
         let OfferOutput { chan, branches } = self;
 
-        // Figure out the name of the `dialectic` crate
-        lazy_static! {
-            static ref CRATE_NAME: String = proc_macro_crate::crate_name("dialectic")
-                .unwrap_or_else(|_| "dialectic".to_owned());
-        }
-        let dialectic_crate = Ident::new(&**CRATE_NAME, Span::call_site());
+        // We need to find the right path where we can reference types in our proc macro. This is a
+        // little tricky. There are three cases to consider.
+        let dialectic_crate: Path = match proc_macro_crate::crate_name("dialectic") {
```

Comment:
This code is duplicated between the `ToTokens` instance for `Spanned<Target>` and this place. Should we unify it somewhere? And if so, where?

Reply:
that seems a little tough to do; the two sections are in completely different crates... i don't know *where* we could put it, maybe in dialectic-compiler? since there's no way for dialectic-compiler to depend on dialectic-macro. unless we made a third tiny crate?